src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CompilationContext.cs

```diff
@@ -842,67 +795,88 @@ private static Imports BuildImports(CSharpCompilation compilation, ImmutableArra
                         {
-                            Debug.Assert(target != null, string.Format("Namespace import string '{0}' has no target", importString));
+                            // DevDiv #999086: Some previous version of VS apparently generated type aliases as "UA{alias} T{alias-qualified type name}". 
+                            // Neither Roslyn nor Dev12 parses such imports.  However, Roslyn discards them, rather than interpreting them as "UA{alias}"
+                            // (which will rarely work and never be correct).
+                            Debug.WriteLine($"Import record '{importRecord}' has syntactically invalid target '{importRecord.TargetString}'");
+                            continue;
+                        }
 
-                            NameSyntax targetSyntax;
-                            if (!SyntaxHelpers.TryParseDottedName(target, out targetSyntax))
+                        NamespaceSymbol namespaceSymbol;
+                        var portableImportRecord = importRecord as PortableImportRecord;
+                        if (portableImportRecord != null)
+                        {
+                            var targetAssembly = portableImportRecord.GetTargetAssembly<ModuleSymbol, AssemblySymbol>(module, module.Module);
+                            if ((object)targetAssembly == null)
+                            {
+                                namespaceSymbol = BindNamespace(binder, targetSyntax);
+                            }
+                            else if (targetAssembly.IsMissing)
                             {
-                                // DevDiv #999086: Some previous version of VS apparently generated type aliases as "UA{alias} T{alias-qualified type name}". 
-                                // Neither Roslyn nor Dev12 parses such imports.  However, Roslyn discards them, rather than interpreting them as "UA{alias}"
-                                // (which will rarely work and never be correct).
-                                Debug.WriteLine("Import string '{0}' has syntactically invalid target '{1}'", importString, target);
+                                Debug.WriteLine($"Import record '{importRecord}' has invalid assembly reference '{targetAssembly.Identity}'");
                                 continue;
                             }
+                            else
+                            {
+                                namespaceSymbol = targetAssembly.GlobalNamespace;
+                                foreach (var name in importRecord.TargetString.Split('.'))
+                                {
+                                    namespaceSymbol = namespaceSymbol.GetMembers(name).SingleOrDefault() as NamespaceSymbol;
 
+                                    if ((object)namespaceSymbol == null)
+                                    {
+                                        break;
+                                    }
+                                }
```

Comment:
Don't we need to call ``` targetSyntax = SyntaxHelpers.PrependExternAlias(externAliasSyntax, targetSyntax); ``` here as well?

Wondering in general why do we need to construct these syntax nodes. Whether it would be possible to modify the binder to directly work with symbols.